src/app/shared/forms/main-form/main-form.component.ts

```diff
@@ -0,0 +1,78 @@
+import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { FormGroup, FormControl, Validators, ValidatorFn, AbstractControl} from '@angular/forms';
+import { Observable } from 'rxjs';
+import { User } from '../../users.model';
+
+@Component({
+  selector: 'app-main-form',
+  templateUrl: './main-form.component.html',
+  styleUrls: ['./main-form.component.css']
+})
+export class MainFormComponent implements OnInit {
+
+  userId: string;
+  @Input() currentUser: Observable<User>;
+
+  mainForm: FormGroup = new FormGroup({
+    firstName: new FormControl('', [Validators.required, Validators.minLength(3), Validators.maxLength(10)]),
+    lastName: new FormControl('', [Validators.required, Validators.minLength(3), Validators.maxLength(10)]),
+    login: new FormControl('', [Validators.required, Validators.minLength(3)]),
+    password: new FormControl('', [Validators.required, Validators.minLength(3), this.passwordValidator(new RegExp('^(?=.*[0-9])(?=.*[a-zA-Z])([a-zA-Z0-9]+)$')).bind(this)]),
+    confirmPassword: new FormControl('', [Validators.required, this.confirmPasswordValidator]),
+    age: new FormControl('', [Validators.required, Validators.maxLength(3)]),
+  });
+
+  @Output() valuesSubmitted: EventEmitter<FormGroup> = new EventEmitter();
+
+  constructor() { }
+
+  ngOnInit(): void {
+    if (this.currentUser) {
+      this.currentUser.subscribe(
+        (user) => {
+          this.userId = user.id;
+          this.mainForm.setValue({
+            firstName: user.firstName,
+            lastName: user.lastName,
+            login: user.login,
+            password: user.password,
+            confirmPassword: '',
+            age: user.age
+          });
+        }
+      )
+    }
+  }
+  userDetails: User;
```

Comment:
This variable should be moved up before any function declarations

Reply:
did this